src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker.cs

```diff
@@ -1606,18 +1611,43 @@ private MethodSymbol InferMethod(BoundCall node, MethodSymbol method, ImmutableA
             // PROTOTYPE(NullableReferenceTypes): Support field initializers in local functions.
         }
 
-        private bool? IsResultNullable(Symbol resultSymbol)
+        private static BoundExpression RemoveImplicitConversions(BoundExpression expr)
         {
-            TypeSymbolWithAnnotations resultType = GetTypeOrReturnTypeWithAdjustedNullableAnnotations(resultSymbol);
+            while (true)
+            {
+                if (expr.Kind != BoundKind.Conversion)
+                {
+                    break;
+                }
+                var conversion = (BoundConversion)expr;
+                if (conversion.ExplicitCastInCode || !conversion.Conversion.Exists)
+                {
+                    break;
+                }
+                expr = conversion.Operand;
```

Comment:
>expr = conversion.Operand; [](start = 16, length = 26)

It feels strange that we are doing this in a loop. Are we supposed to remove only top level conversion?

Reply:
The implicit `Conversion` may have multiple parts (particularly with user-defined conversions) with each part as a separate `BoundConversion` node.

---
In reply to: [169855523](https://github.com/dotnet/roslyn/pull/24935#discussion_r169855523) [](ancestors = 169855523)

Reply:
I think we should target these cases specifically, i.e. recognize them and stop when appropriate rather than zipping through. Consider adding a PROTOTYPE comment
 

---
In reply to: [170122759](https://github.com/dotnet/roslyn/pull/24935#discussion_r170122759) [](ancestors = 170122759,169855523)

Reply:
Add `// PROTOTYPE` comment.